internal/cli/setup/cluster_config.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package setup
+
+import (
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/pointer"
+	atlasClustersPinned "go.mongodb.org/atlas-sdk/v20240530005/admin"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20250312003/admin"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	defaultNodeCount = 3
+	defaultPriority  = 7
+)
+
+func defaultDiskSizeGB(provider, tier string) float64 {
+	return atlas.DefaultDiskSizeGB[strings.ToUpper(provider)][tier]
+}
+
+// newCluster creates a new cluster for the pinned API version.
+func (opts *Opts) newCluster() *atlasClustersPinned.AdvancedClusterDescription {
```

Comment:
this is from `cluster_setup.go`, I updated it a bit to simplify but added unit tests to make sure it's behaving the same